src/Core/Domain/Hook/QueryResult/Hook.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Hook\QueryResult;
+
+class Hook
+{
+    private int $id;
+    private bool $active;
+    private string $name;
+    private string $title;
+    private string $description;
+
+    public function __construct(
+        int $id,
+        bool $active,
+        string $name,
+        string $title,
+        string $description
+    ) {
+        $this->id = $id;
+        $this->active = $active;
+        $this->name = $name;
+        $this->title = $title;
+        $this->description = $description;
+    }
```

Comment:
Or simply: (also with `readonly` we are sure to not mutate properties in any kind of way)

```suggestion
    public function __construct(
        private readonly int $id,
        private readonly bool $active,
        private readonly string $name,
        private readonly string $title,
        private readonly string $description
    ) {
    }
```

Reply:
@lartist proposed this but said we should do getters to have the same structure as the other similar files.

Reply:
Yes you can have getters and constructor property promotions @tleon